src/main/java/com/flipkart/foxtrot/client/FoxtrotClient.java

```diff
@@ -1,34 +1,40 @@
 package com.flipkart.foxtrot.client;
 
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicReference;
+import com.flipkart.foxtrot.client.cluster.FoxtrotCluster;
+import com.flipkart.foxtrot.client.selectors.RandomSelector;
+import com.flipkart.foxtrot.client.senders.HttpSyncEventSender;
+import com.flipkart.foxtrot.client.senders.QueuedSender;
+import com.google.common.base.Strings;
+
+import java.util.List;
 
 public class FoxtrotClient {
-    private final MemberSelector selector;
-    private ScheduledExecutorService executorService;
-    private AtomicReference<FoxtrotClusterStatus> status = new AtomicReference<>();
-
-    public FoxtrotClient(FoxtrotClientConfig config, MemberSelector selector) throws Exception {
-        this.selector = selector;
-        executorService = Executors.newScheduledThreadPool(1);
-        ClusterStatusUpdater updater = ClusterStatusUpdater.create(config, status);
-        executorService.scheduleWithFixedDelay(updater, 0, config.getRefreshInterval(), TimeUnit.SECONDS);
+    private final FoxtrotCluster foxtrotCluster;
+    private final EventSender eventSender;
+
+    public FoxtrotClient(FoxtrotClientConfig config) throws Exception {
+        this.foxtrotCluster = new FoxtrotCluster(config);
+        this.eventSender = (Strings.isNullOrEmpty(config.getLocalQueuePath())
```

Comment:
Client does not care really, if (s)he's passing a path, it means (s)he's expecting it to be persisted. Otherwise not. We can discuss on this if required.